Zend/zend_compile.c

```diff
@@ -5654,6 +5671,10 @@ static zend_type zend_compile_typename(
 			ZSTR_VAL(type_str));
 	}
 
+	if ((type_mask == MAY_BE_ANY) && allow_null) {
```

Comment:
```suggestion
	if (type_mask == MAY_BE_ANY && allow_null) {
```

Reply:
Also, this is probably broken if force_allow_null is set. Please add a test with `mixed $arg = null`.